server/src/main/java/org/elasticsearch/action/admin/indices/stats/IndicesStatsRequest.java

```diff
@@ -54,6 +54,21 @@ public IndicesStatsRequest clear() {
         return this;
     }
 
+    /**
+     * Returns the underlying stats flags.
+     */
+    public CommonStatsFlags flags() {
+        return flags;
+    }
+
+    /**
+     * Sets the underlying stats flags.
+     */
+    public IndicesStatsRequest flags(CommonStatsFlags flags) {
```

Comment:
I looked into removing the redundant accessor methods below, but this made `IndicesStatsRequest` less comfortable to work with. (Note that these request objects are being used in the high-level REST client, so it will be more common for consumers to work with them directly).

Reply:
does it make sense to remove the setters? I imagine it feels more ergonomic to use the `IndicesStatsRequestBuilder` for building up a modified `IndicesStatsRequest`

Reply:
Thanks @talevy! To add more color, `*Request` objects generally support being constructed directly through builder syntax. In some situations where requests are needed, a `*RequestBuilder` can't be created, as it's a fairly rich object that requires an `ElasticsearchClient` as well.

Now that they have a bigger part in one of our clients (the high-level REST client) it'd be great to discuss our approach around `*Request` objects more broadly. The fact they support builder syntax themselves is a departure from the `*RequestBuilder` model, and prevents the requests from being immutable.